File: Java/xunit-test-patterns/src/test/java/_02_basics/Ex01FourPhaseTest.java
package _02_basics;

import lombok.val;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class Ex01FourPhaseTest {

    @Test
    public void shouldAllowAccessingItsName() {
        //GIVEN
        val anyName = "Zenek";
        val user = new User(anyName, "Ziomal");

        //WHEN
        val name = user.getName();

        //THEN
        assertThat(name).isEqualTo(anyName);
    }
    //TODO where is the 4th phase?

    @Test
    public void shouldAllowAccessingItsName2() {
        //we don't use this convention
        //ARRANGE
        val anyName = "Zenek";
        val user = new User(anyName, "Ziomal");

        //ACT
        val name = user.getName();

        //ASSERT
        assertThat(name).isEqualTo(anyName);

        //ANNIHILATE
        //...
    }
}

File: CSharp/XUnitTestPatterns/XUnitTestPatterns/_02_Basics/Ex01FourPhaseTest.cs
using NUnit.Framework;

namespace XUnitTestPatterns._02_Basics
{
  public class Ex01FourPhaseTest
  {
    [Test]
    public void ShouldAllowAccessingItsName()
    {
      //GIVEN
      var anyName = "Zenek";
      var user = new User(anyName, "Ziomal");

      //WHEN
      var name = user.Name;

      //THEN
      Assert.AreEqual(anyName, name);
    }

    //TODO where is the 4th phase?

    [Test]
    public void ShouldAllowAccessingItsName2()
    {
      //we don't use this convention
      //ARRANGE
      var anyName = "Zenek";
      var user = new User(anyName, "Ziomal");

      //ACT
      var name = user.Name;

      //ASSERT
      Assert.AreEqual(anyName, name);

      //ANNIHILATE
      //...
    }
  }

  public class User
  {
    public User(string name, string surname)
    {
      Name = name;
      Surname = surname;
    }

    public string Surname { get; }
    public string Name { get; }
  }
}